packages/grpc-js/src/call-stream.ts

```diff
@@ -377,6 +383,7 @@ export class Http2CallStream extends Duplex implements Call {
   }
 
   cancelWithStatus(status: Status, details: string): void {
+    this.closeChannel();
```

Comment:
I don't know what you think this will do, but it's not correct. There can be multiple calls on a single channel, and a call ending with an error should not close its channel.